tests/test_unit/test_napari_plugin/test_poses_loader_widget.py

```diff
@@ -167,3 +168,28 @@ def test_on_load_clicked_with_valid_file_path(
     # Check that a Points layer was added to the viewer
     points_layer = poses_loader_widget.viewer.layers[0]
     assert points_layer.name == f"poses: {file_path.name}"
+
+
+def test_dimension_slider_matches_frames(make_napari_viewer_proxy):
+    """Test that the dimension slider is set to the correct value when
+    data with NaNs is loaded.
+    """
+    viewer = make_napari_viewer_proxy()
+    poses_loader_widget = PosesLoader(viewer)
+
+    # Set the file path to a file
+    file_path = pytest.DATA_PATHS.get("SLEAP_two-mice_octagon.analysis.h5")
+    poses_loader_widget.file_path_edit.setText(file_path.as_posix())
+    poses_loader_widget.source_software_combo.setCurrentText("SLEAP")
+
+    # Check the data contains nans
+    ds = load_poses.from_file(file_path, "SLEAP", fps=1)
+    assert ds.position.isnull().any()
+
+    # Call the _on_load_clicked method (pretend the user clicked "Load")
+    poses_loader_widget._on_load_clicked()
+
+    # Check the frame slider is set to the correct value
+    assert viewer.dims.range[0] == RangeTuple(
+        start=0.0, stop=ds.position.shape[0] - 1, step=1.0
+    )
```

Comment:
Just a clarification here:

What would happen if say the very first or the very last frame contained only NaNs? My understanding of the code tells me that the frame would be completely dropped, causing the min-max range of frames to change, making this assertion to fail.

It's not the case here because this edge case doesn't manifest in the dataset you chose for this test.

Perhaps we should think of a more comprehensive test to check for this, but we shouldn't do it here in this PR, because I'm keen to get this bugfix merged soon.

Reply:
oh yes good point, can't believe I missed that!

I'd actually say let's not rush it and comprehensively test this in this PR